Exiled.Events/Patches/Events/Player/Spawning.cs

```diff
@@ -37,38 +34,85 @@ private static MethodInfo TargetMethod()
             return Method(TypeByName("PlayerRoles.FirstPersonControl.Spawnpoints.RoleSpawnpointManager").GetNestedTypes(all)[1], "<Init>b__2_0");
         }
 
-        private static bool Prefix(ReferenceHub hub, PlayerRoleBase prevRole, PlayerRoleBase newRole)
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
         {
-            if (newRole.ServerSpawnReason != RoleChangeReason.Destroyed && Player.TryGet(hub, out Player player))
-            {
-                Vector3 oldPosition = hub.transform.position;
-                float oldRotation = (prevRole as IFpcRole)?.FpcModule.MouseLook.CurrentVertical ?? 0;
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            int index = newInstructions.FindLastIndex(x => x.opcode == OpCodes.Ldarg_1);
+
+            Label skipLabel = generator.DefineLabel();
 
-                if (newRole is IFpcRole fpcRole)
+            newInstructions[0].labels.Add(skipLabel);
+
+            newInstructions.InsertRange(
+                0,
+                new CodeInstruction[]
                 {
-                    if (newRole.ServerSpawnFlags.HasFlag(RoleSpawnFlags.UseSpawnpoint) && fpcRole.SpawnpointHandler != null && fpcRole.SpawnpointHandler.TryGetSpawnpoint(out Vector3 position, out float horizontalRot))
-                    {
-                        oldPosition = position;
-                        oldRotation = horizontalRot;
-                    }
+                    // if (newRole is IFpcRole)
+                    //  goto skipLabel
+                    new(OpCodes.Ldarg_3),
+                    new(OpCodes.Isinst, typeof(IFpcRole)),
+                    new(OpCodes.Brtrue_S, skipLabel),
+
+                    // Player.Get(hub)
+                    new(OpCodes.Ldarg_1),
+                    new(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+                    new(OpCodes.Dup),
+
+                    // Player::Position
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(Player), nameof(Player.Position))),
+
+                    // 0f
+                    new(OpCodes.Ldc_R4, 0f),
 
-                    SpawningEventArgs ev = new(player, oldPosition, oldRotation, prevRole);
+                    // oldRole
+                    new(OpCodes.Ldarg_2),
 
-                    Handlers.Player.OnSpawning(ev);
+                    // SpawningEventArgs(Player, Vector3, 0, PlayerRoleBase)
+                    new(OpCodes.Newobj, GetDeclaredConstructors(typeof(SpawningEventArgs))[0]),
 
-                    hub.transform.position = ev.Position;
-                    fpcRole.FpcModule.MouseLook.CurrentHorizontal = ev.HorizontalRotation;
-                    hub.connectionToClient.Send(new FpcOverrideMessage(ev.Position, ev.HorizontalRotation), 0);
```

Comment:
Where is this call ?

Reply:
if you can verify than ev.HorizontalRotation work without this ?

Reply:
I don't really think that it's neccecary cuz game is also changing it (bruh yes, I completly forgot that game will override it again)

Reply:
yeah but wasn't working with base game that why i ask